### PR TITLE
CORE-10689 - Remove signature table

### PR DESCRIPTION
**Pull request by nikinagy**
Remove signature table and move signature information into registration request and member info database tables.

Rumtime-os PR: https://github.com/corda/corda-runtime-os/pull/3454

### CODE REVIEW

File: data/avro-schema/src/main/resources/avro/net/corda/data/membership/common/RegistrationStatusDetails.avsc [hunk 1]
```diff
@@ -40,6 +40,16 @@
...
+    {
+      "name": "memberSignature",
+      "doc": "Signature of the member over the memberContext.",
+      "type": "net.corda.data.crypto.wire.CryptoSignatureWithKey"
+    },
+    {
+      "name": "memberSignatureSpec",
+      "doc": "Signature spec of member signature.",
+      "type": "net.corda.data.crypto.wire.CryptoSignatureSpec"
+    },
```
Comment (charlieR3) [inline data/avro-schema/src/main/resources/avro/net/corda/data/membership/common/RegistrationStatusDetails.avsc:52]: Is the signature relevant to the status of the registration?

File: data/avro-schema/src/main/resources/avro/net/corda/data/membership/PersistentSignedMemberInfo.avsc [hunk 2]
```diff
@@ -0,0 +1,23 @@
+{
+  "type": "record",
+  "name": "PersistentSignedMemberInfo",
+  "namespace": "net.corda.data.membership",
+  "doc": "Avro representation of a member's signature and persistent MemberInfo.",
+  "fields": [
+    {
+      "name": "persistentMemberInfo",
+      "doc": "Persistent member info for the member.",
+      "type": "net.corda.data.membership.PersistentMemberInfo"
```
Comment (charlieR3) [inline data/avro-schema/src/main/resources/avro/net/corda/data/membership/PersistentSignedMemberInfo.avsc:10]: Is it possible to add the signature to `PersistentMemberInfo` or do we have to have a new schema?

Comment (charlieR3) [reply data/avro-schema/src/main/resources/avro/net/corda/data/membership/PersistentSignedMemberInfo.avsc:10]: Actually, `PersistentMemberInfo` has the member context as a KeyValuePairList and not the byte array that the member signed over. So this just keeps the signature and drops the signed data so signature verification could fail as a result.

File: data/avro-schema/src/main/resources/avro/net/corda/data/membership/common/RegistrationStatusDetails.avsc (hunk 1 above)
Comment (nikinagy) [reply data/avro-schema/src/main/resources/avro/net/corda/data/membership/common/RegistrationStatusDetails.avsc:52]: Renamed this to RegistrationRequestDetails in [d2079a0](https://github.com/corda/corda-api/pull/990/commits/d2079a07d1763db59b7031adf2f7f8b808bb3fc2) for better reference of the usage scenarios

File: data/avro-schema/src/main/resources/avro/net/corda/data/membership/PersistentSignedMemberInfo.avsc (hunk 2 above)
Comment (nikinagy) [reply data/avro-schema/src/main/resources/avro/net/corda/data/membership/PersistentSignedMemberInfo.avsc:10]: Left a comment under https://r3-cev.atlassian.net/browse/CORE-11990 to revisit this topic.

**Review by charlieR3 — DISMISSED** [review]

**Review by charlieR3 — APPROVED** [review]

**Pull request merged**